components/api/src/error.rs

```diff
@@ -24,7 +24,7 @@ use serde::{Deserialize, Serialize};
 use std::fmt;
 
 /// API HTTP error struct.
-#[derive(Fail, Debug, Default)]
+#[derive(Fail, Debug, Default, Eq, PartialEq)]
```

Comment:
I'd suggest to **not** implement `PartialEq` / `Eq` for types in which equality may become a problem in the future (i.e., it's not obvious that type instances should be comparable). I've described the reasoning in #1696 - the implementation may be difficult to maintain if new fields are added to the error.

Reply:
I'm wondering when will we be able to use negative traits on custom traits...

```rust
impl !PartifalEq for OurType {}
```